ERCS/erc-7700.md

```diff
@@ -0,0 +1,549 @@
+) authorised(node) {
+    // Route to database storage
+    revert StorageRoutedToDatabase(
+        gatewayUrl
+    );
+}
+
+/**
+* Sets the avatar text record associated with an ENS node
+* [!] May only be called by the owner or manager of that node in ENS registry
+* @param node Namehash of ENS domain to update
+* @param key Key for ENS text record
+* @param value URL to avatar
+*/
+function setText(
+    bytes32 node,
+    string key,
+    string value
+) external {
+    // Verify owner or manager permissions
+    require(authorised(node), "NOT_ALLOWED");
+    // Route to L2 storage
+    revert StorageRoutedToL2(
+        chainId, 
+        contractL2
+    );
+}
+```
+
+#### L2 Contract
+```solidity
+// Function in L2 contract
+function setText(
+    bytes32 node,
+    bytes32 key,
+    bytes32 value
+) external {
+    // Store record mapped by node & sender
+    records[keccak256(abi.encodePacked(node, msg.sender))]["text"][key] = value;
+}
+```
+
+#### Client-side Code
+```ts
+/* Client-side pseudo-code in ENS App */
+// Deterministically generate signer keypair
+let signer = keygen(username, sigKeygen, password);
+// Construct POST body by signing calldata with derived private key
+let post: Post = signData(node, addr, signer.priv);
+// POST to gateway
+await fetch(gatewayUrl, {
+  method: "POST",
+  body: JSON.stringify(post)
+});
+```
+
+## Rationale
+Technically, the cases of L2s and databases are similar; routing to an L2 involves routing the `eth_call` to another EVM, while routing to a database can be made by extracting `eth_sign` from `eth_call` and posting the resulting signature explicitly along with the data for later verification. Methods in this document perform these precise tasks when routing storage operations to external routers. In addition, methods such as signing data with a derived signer (for databases) allow for significant UX improvement by fixing the number of signature prompts in wallets to 2, irrespective of the number of data instances to sign per node or the total number of nodes to update. This improvement comes at no additional cost to the user and allows services to perform batch updates.
+
+## Backwards Compatibility
+1. `StorageRoutedToL2()` is backward compatible and identical to `StorageHandledByL2()` in [EIP-5559](./eip-5559).
+
+2. `StorageRoutedToDatabase()` is not backward compatible with `StorageHandledByDatabase()` in [EIP-5559](./eip-5559).
+
+## Security Considerations
+1. Clients must purge the derived signer private keys from local storage immediately after signing the off-chain data.
+
+2. Signature message payload and the resulting deterministic signature `sigKeygen` must be treated as a secret by the clients and immediately purged from local storage after usage in the `keygen()` function.
+
+3. Clients must immediately purge the `password` from local storage after usage in the `keygen()` function.
+
+## Copyright
+Copyright and related rights waived via [`CC0`](../LICENSE.md).
```

Comment:
```suggestion
Copyright and related rights waived via [CC0](../LICENSE.md).
```

Use the exact text from the template.

Reply:
Fixed in https://github.com/ethereum/ERCs/pull/404/commits/363fc77d6fbba3919a18aaab2ba757e9af9fae42